File: doc/codes/python/examples/api/link/ethercrab.py
from autd3 import Duration

import autd3_link_ethercrab as ethercrab

interface = "eth0"
sync0_period = Duration.from_millis(1)
sync0_shift = Duration.from_millis(0)
sync_tolerance = Duration.from_micros(1)
sync_timeout = Duration.from_secs(10)
# ANCHOR: api
ethercrab.EtherCrabLinkOption(
    interface,
    sync0_period,
    sync0_shift,
    sync_tolerance,
    sync_timeout,
)
# ANCHOR_END: api

File: doc/codes/csharp/examples/api/link/ethercrab.cs
using System;
using AUTD3;
using AUTD3.Link;

namespace DocSamples.ApiLinkEthercrab;

internal static class Sample
{
    internal static void Run()
    {
        var iface = Interface.Auto;
        var sync0Period = TimeSpan.FromMilliseconds(1);
        var sync0Shift = TimeSpan.FromMilliseconds(0);
        var syncTolerance = TimeSpan.FromMicroseconds(1);
        var syncTimeout = TimeSpan.FromSeconds(10);
        // ANCHOR: api
        new EtherCrabLinkOption(
            iface,
            sync0Period,
            sync0Shift,
            syncTolerance,
            syncTimeout
        );
        // ANCHOR_END: api

        var pduTimeout = TimeSpan.FromMilliseconds(30);
        var stateTransitionTimeout = TimeSpan.FromSeconds(10);
        var txRxAffinity = 0UL;
        // ANCHOR: api_extra
        new EtherCrabLinkOption(
            iface,
            sync0Period,
            sync0Shift,
            syncTolerance,
            syncTimeout,
            pduTimeout,
            stateTransitionTimeout,
            txRxPriority: 90,
            disableTxRxPriority: false,
            txRxPolicy: RtSchedulePolicy.Fifo,
            txRxAffinity: txRxAffinity
        );
        // ANCHOR_END: api_extra
    }
}
